### PR TITLE
clippy: Fix some warnings in `script/dom`

### PR DESCRIPTION
**Pull request by ektuu**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are a part of https://github.com/servo/servo/issues/31500
- [x] These changes do not require tests because only clippy warnings were fixed.

### CODE REVIEW

File: components/script/dom/texttracklist.rs [hunk 1]
```diff
@@ -125,7 +124,7 @@ impl TextTrackListMethods for TextTrackList {
         self.dom_tracks
             .borrow()
             .iter()
-            .filter(|track| track.id() == &id_str)
+            .filter(|track| track.id() == id_str)
             .next()
```
Comment (mukilan) [inline components/script/dom/texttracklist.rs:128]: Is there a reason why this particular `.filter(_).next()` chain is not replaced with `.find`?

Comment (ektuu) [reply components/script/dom/texttracklist.rs]: It seems that I overlooked this piece of code, I apologise for the mistake.
It has been fixed now!

**Review by mrobinson — APPROVED** [review]

File: components/script/dom/texttracklist.rs [hunk 2]
```diff
@@ -125,8 +124,8 @@ impl TextTrackListMethods for TextTrackList {
         self.dom_tracks
             .borrow()
             .iter()
-            .filter(|track| track.id() == id_str)
-            .next()
+            .find(|track| track.id() == id_str)
+
```
Comment (mrobinson) [inline components/script/dom/texttracklist.rs:128]:
```suggestion
```

**Review by mrobinson — CHANGES_REQUESTED** [review]

**Review by mrobinson — APPROVED** [review]

**Comment by ektuu** [comment]
Since auto merge has been enabled but WPT Export / upstream (pull_request_target) failed unexpectedly, what should I do?

**Comment by mrobinson** [comment]
@ektuu I think you need to rebase this change to remove all the merge commits and then do a force push to your remote branch.

**Comment by ektuu** [comment]
> @ektuu I think you need to rebase this change to remove all the merge commits and then do a force push to your remote branch.

done!
Thanks.

**Pull request merged**